routes/productRoute.js

```diff
@@ -8,12 +8,15 @@ const {
   getProduct,
 } = require('../controllers/productController');
 const validateSchema = require('../middlewares/ValidateSchema');
-const {addProductSchema} = require('../validations/products/addProduct');
+const {
+  addProductSchema,
+  getProductSchema,
+} = require('../validations/products');
 
 route.post('/', validateSchema(addProductSchema), addproduct);
 
 route.get('/', getAllProducts);
 
-route.get('/:id', getProduct);
+route.get('/:id', validateSchema(getProductSchema), getProduct);
 
```

Comment:
if you test the schema I mentioned below and did not work , maybe we will start using package to validate schema:
const { validate } = require('express-validation');
route('/:id',validate(getProductSchema,{context : true}),getProduct);